llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -6751,6 +6802,32 @@ Value *SPIRVToLLVM::transSPIRVImageWriteFromInst(SPIRVInstruction *bi, BasicBloc
   addr[lgc::Builder::ImageAddressIdxCoordinate] = transValue(bii->getOpValue(opndIdx++), bb->getParent(), bb);
   Value *texel = transValue(bii->getOpValue(opndIdx++), bb->getParent(), bb);
 
+  // Create smaller texel vector, if num texel components > num channels. This
+  // ensures that no unused channels are being stored.
+  if (auto texelVecType = dyn_cast<FixedVectorType>(texel->getType())) {
+    unsigned numChannels = getImageNumChannels(imageInfo.desc);
+    if (numChannels > 0 && numChannels < texelVecType->getNumElements()) {
+      if (numChannels == 1) {
+        texel = getBuilder()->CreateExtractElement(texel, getBuilder()->getInt32(0));
+      } else {
```

Comment:
You could simplify the series of extracts/inserts by using shufflevector instead.

Reply:
Thanks. Will change.